torch/csrc/jit/tensorexpr/schedule.cpp

```diff
@@ -392,6 +392,203 @@ ScheduleObject* ScheduleNode::CloneScheduleObject(ScheduleObject* object) {
   return new_object;
 }
 
+class Vectorizer : public IRMutator {
+  public:
+  Vectorizer(const Var& v, int lanes) : var_(v), lanes_(lanes) {}
+
+  Expr mutate(const Add* v) override {
+    std::vector<Expr> inputs = { v->lhs(), v->rhs() };
+    return try_vectorize(Expr(v), inputs,
+      [&](){ return inputs[0] + inputs[1]; });
+  }
+
+  Expr mutate(const Sub* v) override {
+    std::vector<Expr> inputs = { v->lhs(), v->rhs() };
+    return try_vectorize(Expr(v), inputs,
+      [&](){ return inputs[0] - inputs[1]; });
+  }
+
+  Expr mutate(const Mul* v) override {
+    std::vector<Expr> inputs = { v->lhs(), v->rhs() };
+    return try_vectorize(Expr(v), inputs,
+      [&](){ return inputs[0] * inputs[1]; });
+  }
+
+  Expr mutate(const Div* v) override {
+    std::vector<Expr> inputs = { v->lhs(), v->rhs() };
+    return try_vectorize(Expr(v), inputs,
+      [&](){ return inputs[0] / inputs[1]; });
+  }
+
+  Expr mutate(const Max* v) override {
+    std::vector<Expr> inputs = { v->lhs(), v->rhs() };
+    return try_vectorize(Expr(v), inputs,
+      [&](){ return Max::make(inputs[0], inputs[1], v->propagate_nans()); });
+  }
+
+  Expr mutate(const Min* v) override {
+    std::vector<Expr> inputs = { v->lhs(), v->rhs() };
+    return try_vectorize(Expr(v), inputs,
+      [&](){ return Min::make(inputs[0], inputs[1], v->propagate_nans()); });
+  }
+
+  Expr mutate(const CompareSelect* v) {
+    std::vector<Expr> inputs = { v->lhs(), v->rhs() };
+    return try_vectorize(Expr(v), inputs,
+      [&](){ return CompareSelect::make(inputs[0], inputs[1], v->compare_select_op()); });
+  }
+
+  Expr mutate(const Cast* v) {
+    std::vector<Expr> inputs = { v->src_value() };
+    return try_vectorize(Expr(v), inputs,
+      [&](){ return Cast::make(Dtype(v->dtype().scalar_type(), lanes_), inputs[0]); });
+  }
+
+  Expr mutate(const Variable* v) override {
+    if (v == var_.AsNode<Variable>()) {
+      Expr ret = Expr(v) * lanes_;
+      vectorized.insert(ret.AsNode<BaseExprNode>());
+      ret = Ramp::make(ret, 1, lanes_);
+      vectorized.insert(ret.AsNode<BaseExprNode>());
+      return ret;
+    }
+
+    return Expr(v);
+  }
+
+  Expr mutate(const Let* v) {
+    Expr var = v->var();
+    Expr value = v->value();
+    Expr body = v->body();
+
+    std::vector<Expr> inputs = { body };
```

Comment:
Minor nit: probably a good idea for us to add a generic inputs(...), outputs(...) function for all operators.